cmake/OpenCVInstallLayout.cmake

```diff
@@ -20,7 +20,7 @@ if(ANDROID)
   ocv_update(OPENCV_JNI_INSTALL_PATH            "${OPENCV_LIB_INSTALL_PATH}")
   ocv_update(OPENCV_JNI_BIN_INSTALL_PATH        "${OPENCV_JNI_INSTALL_PATH}")
 
-elseif(WIN32 AND CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
+elseif(MSVC AND CMAKE_HOST_SYSTEM_NAME MATCHES Windows)
```

Comment:
Compiler check is not suitable here.
Only platform checks should be used here.
Use `NOT MINGW` instead.

Anyway this is a breaking change, so add new extra option to control this behavior (look for `OPENCV_FORCE_*` / `OPENCV_SKIP_*` behavior).

Reply:
May I ask why this is a breaking change?

Reply:
> breaking change

This is a compatibility topic of OpenCV version upgrades.
This means that behavior of already released code is changing by this patch.
In general we should avoid that.

Changes could be be done safe without questions for broken code (which never works) or never released code.

Proposed MinGW layout is already using by OpenCV users. Changing that would break their scripts during OpenCV version upgrade.
E.g. here: https://github.com/huihut/OpenCV-MinGW-Build

if we want to change behavior, then we should left option for users which don't want or cant migrate on new behavior now (we also need a short note for "Changelog" in PRs description).
This is necessary for smooth OpenCV upgrade process.